README.md

```diff
@@ -25,11 +25,25 @@ const cqlfhir = require('cql-exec-fhir');
 // Code setting up the CQL library, executor, etc, and getting the patient data as a bundle
 // ...
 
-const patientSource = cqlfhir.PatientSource.FHIRv102(); // or .FHIRv300() or .FHIRv400()
+const patientSource = cqlfhir.PatientSource.FHIRv401(); // or .FHIRv102() or .FHIRv300() or .FHIRv400()
 patientSource.loadBundles([patient01, patient02]);
 const results = executor.exec(patientSource);
 ```
 
+## (Optional) Trusted Environment with meta.profile
+
+If desired, the FHIR Data Source can be configured to use the `meta.profile` list on FHIR resources as a source of truth for whether or not that resource should be included when looking through the Bundle of data.
+
+```js
+const cqlfhir = require('cql-exec-fhir');
+
+// Passing 'true' to the constructor enables the trusted environment
+const patientSource = cqlfhir.PatientSource.FHIRv401(true); // or .FHIRv102() or .FHIRv300() or .FHIRv400()
+```
+
+As an example, if an ELM Retrieve expression asks for a FHIR Resource with profile `http://hl7.org/fhir/StructureDefinition/Condition`, the default behavior of the FHIR Data Source is to find any FHIR Condition resource.
+With the trusted environment enabled however, the FHIR Data Source will _only_ find resources with the string `'http://hl7.org/fhir/StructureDefinition/Condition'` included in its `meta.profile` list.
```

Comment:
Yeah this makes sense. I'll change the example to be something like QI Core or US Core which will be common use cases and make the requested change for the `if` statement as well. 

For context, I think one of the reasons we see things like the base FHIR structure defs in this code is because in our use case, we are calculating eCQMs where the patient data relevant to some eCQM has been submitted to a FHIR server that can do profile-level validation, and will always be tagging the incoming resources with `meta.profile` with the profile(s) that it validated against in QI Core, and I think in base FHIR (at least that's how we implemented it, which could be wrong. We have no examples to go off lol, blazing our own trail).

So I think where we fell short was that in our use case, valid FHIR Conditions that are submitted to our server with validation enabled will always be tagged with `http://hl7.org/fhir/StructureDefinition/Condition` in the `meta.profile` meaning that ok nice we can use that as a source of truth because we know it'll be there. As you've pointed out though, other use cases could/will be different, and very likely will not have the base FHIR structure def tagged on resources all the time even during the trusted environment agreement. 

All this is to say that I agree with you and I think it's safe to not enforce the base FHIR structure defs for this, but hopefully that clarified what we were doing a bit more

Reply:
Pretty much, yeah. We're basically doing all the MAYs and SHOULDs at the end of this snippet in the FHIR docs about `meta.profile` 
![image](https://user-images.githubusercontent.com/16297930/194615069-2156bcb0-b9db-4424-b981-65018364e292.png)

It's not a perfect system, but it's what we agreed on trying out in the short term so that no one asks us to do profile-level validation at runtime ;)

Reply:
> It's not a perfect system, but it's what we agreed on trying out in the short term so that no one asks us to do profile-level validation at runtime ;)

Yeah, you definitely don't want to be in that game!

> here is a slight assumption that the server that we'll be dealing with for our use case will only let it through if the claim is verified

Hence the term "trusted environment".  Got it.